src/compiler/types.ts

```diff
@@ -885,6 +885,144 @@ namespace ts {
         /* @internal */ jsDocCache?: readonly JSDocTag[];     // Cache for getJSDocTags
     }
 
+    /* @internal */
+    export type HasChildren =

```

Comment:
I assume that if we need to update this, we'll know because some other code won't be able to compile?

Reply:
To add a method to `visitEachChild` (and potentially `forEachChild`), we'd have to add the node to this union.